controllers/lock_client.go

```diff
@@ -156,6 +169,72 @@ func (client *RealLockClient) updateLock(transaction fdb.Transaction, start int6
 	transaction.Set(lockKey, lockValue.Pack())
 }
 
+// AddPendingUpgrades registers information about which process groups are
+// pending an upgrade to a new version.
+func (client *RealLockClient) AddPendingUpgrades(version fdbtypes.FdbVersion, processGroupIDs []string) error {
+	_, err := client.database.Transact(func(tr fdb.Transaction) (interface{}, error) {
+		for _, processGroupID := range processGroupIDs {
+			err := tr.Options().SetAccessSystemKeys()
+			if err != nil {
+				return nil, err
+			}
+
+			key := fdb.Key(fmt.Sprintf("%s/upgrades/%s/%s", client.cluster.GetLockPrefix(), version.String(), processGroupID))
+			tr.Set(key, []byte(processGroupID))
+		}
+		return nil, nil
+	})
+	return err
+}
+
+// GetPendingUpgrades returns the stored information about which process
+// groups are pending an upgrade to a new version.
+func (client *RealLockClient) GetPendingUpgrades(version fdbtypes.FdbVersion) (map[string]bool, error) {
+	upgrades, err := client.database.Transact(func(tr fdb.Transaction) (interface{}, error) {
+		err := tr.Options().SetReadSystemKeys()
+		if err != nil {
+			return nil, err
+		}
+
+		keyPrefix := []byte(fmt.Sprintf("%s/upgrades/%s/", client.cluster.GetLockPrefix(), version.String()))
+		keyRange, err := fdb.PrefixRange(keyPrefix)
+		if err != nil {
+			return nil, err
+		}
+		results := tr.GetRange(keyRange, fdb.RangeOptions{}).GetSliceOrPanic()
+		upgrades := make(map[string]bool, len(results))
+		for _, result := range results {
+			upgrades[string(result.Value)] = true
+		}
+		return upgrades, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return upgrades.(map[string]bool), nil
+}
+
+// ClearPendingUpgrades clears any stored information about pending
+// upgrades.
+func (client *RealLockClient) ClearPendingUpgrades() error {
+	_, err := client.database.Transact(func(tr fdb.Transaction) (interface{}, error) {
+		err := tr.Options().SetAccessSystemKeys()
+		if err != nil {
+			return nil, err
+		}
+
+		keyPrefix := []byte(fmt.Sprintf("%s/upgrades/", client.cluster.GetLockPrefix()))
```

Comment:
Shouldn't the prefix be `fmt.Sprintf("%s/upgrades/%s/", client.cluster.GetLockPrefix(), version.String())` version specific like the other prefixes?

Reply:
No, this method isn't scoped to a version.